tsdb/store.go

```diff
@@ -1391,9 +1391,9 @@ func (s *Store) DeleteSeries(database string, sources []influxql.Source, conditi
 	epochs := s.epochsForShards(shards)
 	s.mu.RUnlock()
 
-	// Limit to 1 delete for each shard since expanding the measurement into the list
+	// Limit deletes for each shard since expanding the measurement into the list
 	// of series keys can be very memory intensive if run concurrently.
-	limit := limiter.NewFixed(1)
+	limit := limiter.NewFixed(s.EngineOptions.Config.MaxConcurrentDeletes)
```

Comment:
I think that the slow delete issues we have seen are more likely to be from issues with contention on `waiter` below than from issues with this limiter.

Given that this particular limiter appears to be purely for performance, not correctness (since it is about concurrent operations on different shards, rather than on the same shard, and two different deletes can already get here concurrently) this seems safe enough - much more clearly safe than I expected when we discussed it earlier.